File: src/main/java/BruteForce/boj_15721/Main.java
package bruteforce.boj_15721;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

// 14924kb 112ms
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader((System.in)));
        int people = Integer.parseInt(br.readLine());
        int t = Integer.parseInt(br.readLine());
        String want = br.readLine();
        StringBuilder sb = new StringBuilder();
        String intro = "0101";
        int n = 1;
        while (sb.length() < t*2) {
            sb.append(intro);
            sb.append("0".repeat(n+1)).append("1".repeat(n+1));
            n += 1;
        }
        String[] str = sb.toString().split("");
        int count = 0;
        for (int i=0;i<str.length;i++) {
            if (want.equals(str[i])) {
                count++;
                if (count == t) {
                    System.out.println(i % people);
                }
            }
        }
    }
}

File: src/main/java/BruteForce/boj_15721/Main.kt
package bruteforce.boj_15721

import java.io.BufferedReader
import java.io.InputStreamReader

// 15032kb 96ms
fun main() {
	val br = BufferedReader(InputStreamReader(System.`in`))
	val a = br.readLine().toInt()
	val t = br.readLine().toInt()
	val g = br.readLine()
	// n -> n+1
	val sb = StringBuilder()
	val intro = "0101"
	var n = 1
	while(sb.length < t*2) {
		sb.append(intro)
		sb.append("0".repeat(n+1)).append("1".repeat(n+1))
		n += 1
	}
	var result = 0
	sb.forEachIndexed { idx, it ->
		if (it.toString() == g) {
			result += 1
			if (result == t) {
				println(idx % a)
			}
		}
	}
}